utils/generate_api.py

```diff
@@ -764,6 +764,37 @@ def dump_modules(modules: Any) -> None:
     unasync.unasync_files(filepaths, rules)
     blacken(CODE_ROOT / "opensearchpy")
 
+    # Updating the CHANGELOG.md
+    response = requests.get(
+        "https://api.github.com/repos/opensearch-project/opensearch-api-specification/commits"
+    )
+    if response.ok:
+        commit_info = response.json()[0]
+        commit_url = commit_info["html_url"]
+        latest_commit_sha = commit_info.get("sha")
+    else:
+        print("Failed to fetch opensearch-api-specification commit information.")
+        print("Status code:", response.status_code)
+        latest_commit_sha = None
```

Comment:
I would bail with a hard failure here (raise error). If we fail to get the api spec history something went wrong.